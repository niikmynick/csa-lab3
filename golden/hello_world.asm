.data
    text "Hello, World!"

.code
    push text
    write
    pop

    halt
