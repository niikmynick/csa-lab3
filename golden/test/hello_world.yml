in_source: |-
  .data
    text "Hello, World!\n"
  .code
    push text
    write
    halt
in_stdin: |-
out_bin: |-
  1111111100000000000011100000001101001000011001010110110001101100011011110010110000100000010101110110111101110010011011000110010000100001000010100000000011111110000000010000000100000000000000000000000000000000000000000000000000000000001000010000001100000000000000000000000000000000000000000000000000000000010000000000001100000000000000000000000000000000000000000000000000000000
out_stdout: |-
  Hello, World!
out_log: |-
  INFO      ::     INPUT       Input schedule: []
  INFO      ::     SYSTEM      Starting execution
  DEBUG     ::     SYSTEM      Initial state: ControlUnit: time=0, instructions executed=0
  DEBUG     ::     SYSTEM      Instructions memory: {0: PUSH $2 - 112, 1: WRITE - 213, 2: HALT - 403}
  DEBUG     ::     SYSTEM      Data memory: {0: InputManager, 1: OutputManager, 2: 72, 3: 101, 4: 108, 5: 108, 6: 111, 7: 44, 8: 32, 9: 87, 10: 111, 11: 114, 12: 108, 13: 100, 14: 33, 15: 10, 16: 0}
  DEBUG     ::     INTER       Have 0 interruptions
  DEBUG     ::     SYSTEM      ControlUnit: time=5, instructions executed=0
  INFO      ::     INSTR       Processing PUSH ADDRESS 2
  DEBUG     ::     INTER       Have 0 interruptions
  DEBUG     ::     SYSTEM      ControlUnit: time=28, instructions executed=1
  INFO      ::     INSTR       Processing WRITE
  DEBUG     ::     INTER       Have 1 interruptions
  DEBUG     ::     INTER       Processing Interruption: OUTPUT Hello, World!
  
  DEBUG     ::     SYSTEM      ControlUnit: time=40, instructions executed=2
  INFO      ::     INSTR       Processing HALT
  DEBUG     ::     INTER       Have 1 interruptions
  DEBUG     ::     INTER       Processing Interruption: HALT
  INFO      ::     SYSTEM      Halting
  INFO      ::     SYSTEM      Execution finished, state: ControlUnit: time=47, instructions executed=3
  INFO      ::     INPUT       Turning off input device.
  INFO      ::     OUTPUT      Turning off output device.


